zephyr/CMakeLists.txt

```diff
@@ -427,12 +427,34 @@ zephyr_library_sources_ifdef(CONFIG_COMP_FIR
 )
 
 if(CONFIG_IPC_MAJOR_3)
+	zephyr_library_sources_ifdef(CONFIG_COMP_FIR
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_hifi3.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_hifi2ep.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_generic.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_ipc3.c
+		${SOF_MATH_PATH}/fir_generic.c
+		${SOF_MATH_PATH}/fir_hifi2ep.c
+		${SOF_MATH_PATH}/fir_hifi3.c
+	)
+
 	zephyr_library_sources_ifdef(CONFIG_COMP_IIR
 		${SOF_AUDIO_PATH}/eq_iir/eq_iir.c
 		${SOF_AUDIO_PATH}/eq_iir/eq_iir_ipc3.c
 		${SOF_AUDIO_PATH}/eq_iir/eq_iir_generic.c
 	)
 elseif(CONFIG_IPC_MAJOR_4)
+	zephyr_library_sources_ifdef(CONFIG_COMP_FIR
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_hifi3.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_hifi2ep.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_generic.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir.c
+		${SOF_AUDIO_PATH}/eq_fir/eq_fir_ipc4.c
+		${SOF_MATH_PATH}/fir_generic.c
+		${SOF_MATH_PATH}/fir_hifi2ep.c
+		${SOF_MATH_PATH}/fir_hifi3.c
```

Comment:
This is the same list twice! Why copy/paste/diverge?

Reply:
for separate ipc3/ipc4 code, if you have other good ways, I can have a try.

Reply:
After a second look I see only one file difference. So only that file should be under CONFIG_IPC_. No copy/paste required for the others.